src/main/kotlin/com/jetbrains/snakecharm/lang/psi/types/AbstractSmkRuleOrCheckpointType.kt

```diff
@@ -77,6 +81,40 @@ abstract class AbstractSmkRuleOrCheckpointType<T : SmkRuleOrCheckpoint>(
 
     override fun isBuiltin() = false
 
+    protected open fun getUseSections(name: String, location: PyExpression): List<RatedResolveResult> {
+        val result = mutableListOf<RatedResolveResult>()
+        if (location.parent is SmkUse) {
```

Comment:
could we use `containgRule is SmkUse` here? We already have containing rule field. Not sure that `location.parent` is always containing rule for all use cases.

Reply:
I think that it's better to use `location.parent`, because here we want to ignore only the module reference in `use` section, and do not want to ignore e.g. references inside `USE_IMPORTED_RULES_NAMES`, but both references would have `SmkUse` as `containingRule`, so in case of module reference we want to check type of the first parent. Also the `containingRule` is `null` when it is called from `SmkRuleOrCheckpointNameReference` so that's another reason, why I prefere `location.parent`

Reply:
Ok. I've added a comment with this explanation